pkg/subroutines/extensions.go

```diff
@@ -157,34 +164,69 @@ func collectExtensions(ctx context.Context, cl client.Client, lookupNamespace st
 	return extensions, nil
 }
 
-func getParentAccount(ctx context.Context, cl client.Client, ns string) (*v1alpha1.Account, error) {
+func getParentAccount(ctx context.Context, cl client.Client, ns string) (*v1alpha1.Account, *string, error) {
+
+	if cluster, ok := kontext.ClusterFrom(ctx); ok && !cluster.Empty() {
+
+		wsCtx := kontext.WithCluster(ctx, logicalcluster.Name(""))
+		list := &tenancyv1alpha1.WorkspaceList{}
+
+		err := cl.List(wsCtx, list)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		for _, ws := range list.Items {
+			if ws.Spec.Cluster != cluster.String() {
+				continue
+			}
+
+			clusterName := ws.Annotations[logicalcluster.AnnotationKey]
+
+			parentCtx := kontext.WithCluster(ctx, logicalcluster.Name(clusterName))
+
+			parentAccount := v1alpha1.Account{}
+			err = cl.Get(parentCtx, types.NamespacedName{
+				Name:      ws.Annotations[v1alpha1.NamespaceAccountOwnerLabel],
```

Comment:
and for now the assumption would be, that this annotation made its way somehow to the workspace and we can add this logic in a later PR or configuration

Reply:
yes, probably this annotation could find its way through the extension mechanism of the organization level accounts. There we could define a metadata section with this (and potentially the namespace annotation) information based on the account which created it.